packages/front-end/components/Metrics/DateGraph.tsx

```diff
@@ -277,54 +209,181 @@ const DateGraph: FC<{
     });
   }
 
+  const { containerRef, containerBounds } = useTooltipInPortal({
+    scroll: true,
+    detectBounds: true,
+  });
+
+  const width = (containerBounds?.width || 0) + margin[1] + margin[3];
+  const dateNums = data.map((d) => getValidDate(d.d).getTime());
+  const min = Math.min(...dateNums);
+  const max = Math.max(...dateNums);
+  const yMax = height - margin[0] - margin[2];
+  const xMax = containerBounds?.width || 0;
+  const numXTicks = width > 768 ? 7 : 4;
+  const numYTicks = 5;
+  const axisHeight = 30;
+  const minGraphHeight = 100;
+  const expBarHeight = 10;
+  const expBarMargin = 4;
+  const expHeight = bands.size * (expBarHeight + expBarMargin);
+  let graphHeight = yMax - expHeight;
+  if (graphHeight < minGraphHeight) {
+    height += minGraphHeight - (yMax - expHeight);
+    graphHeight = minGraphHeight;
+  }
+  const xScale = scaleTime({
+    domain: [min, max],
+    range: [0, xMax],
+    round: true,
+  });
+  const yScale = scaleLinear<number>({
+    domain: [
+      0,
+      Math.max(
+        ...data.map((d) =>
+          type === "binomial" ? d.c : Math.min(d.v * 2, d.v + (d.s ?? 0) * 2)
+        )
+      ),
+    ],
+    range: [graphHeight, 0],
+    round: true,
+  });
+
+  useEffect(() => {
+    if (!hoverDate) {
+      hideTooltip();
+      return;
+    }
+    const datapoint = getDatapointFromDate(hoverDate);
+    if (!datapoint) {
+      hideTooltip();
+      return;
+    }
+    const tooltipData = getTooltipDataFromDatapoint(datapoint);
+    if (!tooltipData) {
+      hideTooltip();
+      return;
+    }
+    showTooltip({
+      tooltipLeft: tooltipData.x,
+      tooltipTop: tooltipData.y,
+      tooltipData: tooltipData,
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [hoverDate]);
```

Comment:
Exhaustive deps checks generally help us prevent bugs. Which deps are we not passing in, and why not?

Reply:
It's 'getDatapointFromDate', 'getTooltipDataFromDatapoint', 'hideTooltip', and 'showTooltip'. These are all pre-defined `const` methods. I tend to prefer listing the dependencies that will effect the outcome of the hook to make it clearer to the reader, but if we prefer listing everything we can do that too.

Reply:
Logically, if any of those functions change, it should result in different logic, right? Are those functions memoized with `useCallback`? Do those functions have dependencies? The function could change and affect the outcome if the function's dependencies change. Also if we add a new dependency to this dependency array and the tooling that is designed to help us is disabled, we could introduce a new bug when a new dependency is added. I personally prefer to only disable exhaustive deps checks if I'm depending on a buggy hook that changes too frequently and causes an infinite re-render.

Reply:
I tried using exhaustive dependencies, and it led to quite a rabbit hole of refactoring and some infinite useEffect loops. There is definitely some fragility in there due to visx's width being a side effect of render. I managed to get it all working and either externalize or memoize utility functions. There was another weird quirk where `margin: [number, number, number, number]` in a dep was causing a loop, but `margin[0], margin[1], margin[2], margin[3]` was not causing the loop. Go figure...

Reply:
Nice, glad you got it figured out. 

And yeah, I'm wondering if they do an equals check and not a deep equals check and that's why the array check was failing, e.g. they may be constructing a new array each time with the same values.